pycarwings2/pycarwings2.py

```diff
@@ -118,7 +118,12 @@ def _request(self, endpoint, params):
         else:
             params["custom_sessionid"] = ""
 
-        req = Request('POST', url=BASE_URL + endpoint, data=params).prepare()
+        req = Request(
+            'POST',
+            url=BASE_URL + endpoint,
+            data=params,
```

Comment:
while we're in here (OUT OF SCOPE), but consider:
https://docs.python.org/3/library/urllib.request.html#urllib.request.Request

> An appropriate Content-Type header should be included if the data argument is present. If this header has not been provided and data is not None, Content-Type: application/x-www-form-urlencoded will be added as a default.

I suppose maybe Nissan API isn't caring ... BUT ... technically, shouldn't this be `application/json`? https://www.ietf.org/rfc/rfc4627.txt

Reply:
Only works if you are sending the parameters as JSON instead of a query string. Passing the data as a query string is done "automatically" using the `data=` parameter, JSON data can be forced using `json=`. 

Posting is done as query string to the endpoint, and the answers are in JSON.